model/src/main/java/oracle/kubernetes/weblogic/domain/model/Domain.java

```diff
@@ -460,4 +474,57 @@ public boolean equals(Object other) {
         .append(status, rhs.status)
         .isEquals();
   }
+
+  public List<String> getValidationFailures() {
+    return new Validator().getValidationFailures();
+  }
+
+  class Validator {
+    private List<String> failures = new ArrayList<>();
+    private Set<String> clusterNames = new HashSet<>();
+    private Set<String> serverNames = new HashSet<>();
+
+    List<String> getValidationFailures() {
+      addDuplicateNames();
+      addUnmappedLogHome();
+
+      return failures;
+    }
+
+    private void addDuplicateNames() {
+      getSpec().getManagedServers().stream().map(ManagedServer::getServerName).forEach(this::checkDuplicateServerName);
+      getSpec().getClusters().stream().map(Cluster::getClusterName).forEach(this::checkDuplicateClusterName);
+    }
+
+    private void checkDuplicateServerName(String s) {
+      if (serverNames.contains(s))
+        failures.add(String.format("More than one server is named '%s'", s));
+      else
+        serverNames.add(s);
+    }
+
+    private void checkDuplicateClusterName(String s) {
+      if (clusterNames.contains(s))
+        failures.add(String.format("More than one cluster is named '%s'", s));
```

Comment:
How could this ever happen unless there was a bug in our topology generator?
Regardless 'cluster' --> 'WebLogic Cluster in your WebLogic domain'

Reply:
Note that it _is_ possible for two cluster to have the same name if they both happen to resolve to same DNS1123 form (all lower case, underscores converted to hyphens).  There's a pending JIRA to check this in the introspector, but maybe it'd fine to check it here instead.

Reply:
This is checking the domain resource, not the WLS domain.

Reply:
ok

Reply:
Actually, not 100% OK. It'd be helpful if the error message clearly stated that this is a domain resource issue, and maybe which specific part of the schema was messed up.